### DISCUSSION TITLE
포스트 프로세싱 DOF 값을 설정하고 싶은데 TypeScript에서 어떻게 접근해야 하나요?

### DISCUSSION THREAD

**Discussion by archistnine** [naverz/zepeto-studio-kor#977] [5. Q&A]
import { Physics, Ray, RaycastHit, Vector3 } from 'UnityEngine';
import { DepthOfField, PostProcessEffectSettings, PostProcessVolume } from 'UnityEngine.Rendering.PostProcessing';
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'

export default class DOFcontroller extends ZepetoScriptBehaviour
{

   private raycast: Ray;
   private raycastHit: RaycastHit;
   private hitInfo: RaycastHit;

   private isHit: bool;
   private hitDistance: number;

   public maxDistance: number;

   private volume: PostProcessVolume;
   private depthOfField: DepthOfField;

   Start()
   {
      // DOF 값을 설정하고 싶은데 TypeScript에서 어떻게 접근해야 하나요? 
      this.volume.profile.TryGetSettings(out depthOfField);  // unity
   }

   Update()
   {
      this.raycast = new Ray(this.transform.position, this.transform.forward * this.maxDistance);

      this.isHit = false;

      const _out = $ref<RaycastHit>();

      if (Physics.Raycast(this.raycast, _out, this.maxDistance))
      {
         const hitInfo = $unref(_out);

         this.isHit = true;
         this.hitDistance = Vector3.Distance(this.transform.position, hitInfo.point);
      }
      else
      {
         if (this.hitDistance < this.maxDistance)
         {
            this.hitDistance++;
         }
      }

      this.SetFocus();
   }

   private SetFocus()
   {
      // 여기에서 각 값을 조정하려 하는데  오류가 나는데 TypeScript에서 어떻게 접근해야 하나요?
      this.depthOfField.focusDistance.value = this.hitDistance; //unity
   }
}

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 포스트프로세싱  DOF 는 아래와 같이 가져올 수 있습니다. (zepeto.world@1.5.0 업데이트 필요)

```typescript
//
this.volume.profile.GetSetting<DepthOfField>();

//
let dofRef = new $ref<DepthOfField>();
const result = this.volume123.profile.TryGetSettings<DepthOfField>(dofRef);
var dof = $unref<DepthOfField>(refDot);

```

**Reply by archistnine** [reply]
감사합니다.

**Reply by archistnine** [reply]
Dof는 접근가능한데 프러퍼티를 조정하려니 value에 오류가 나는데  접근하는 방법이 없을까요?
this.dof.focusDistance.value
